backend/src/main/java/saci/application/model/ScoreResponse.java

```diff
@@ -0,0 +1,87 @@
+package saci.application.model;
+
+import java.util.Optional;
+import saci.domain.model.Level;
+
+public class ScoreResponse {
+    private Optional<LevelResponse> currentLevel;
+    private Optional<LevelResponse> nextLevel;
+
+    private ScoreResponse(Builder builder) {
+        this.currentLevel = builder.currentLevel;
+        this.nextLevel = builder.nextLevel;
+    }
+
+    public Optional<LevelResponse> getCurrentLevel() {
+        return currentLevel;
+    }
+
+    public Optional<LevelResponse> getNextLevel() {
+        return nextLevel;
+    }
+
+    public static class Builder {
```

Comment:
Instead of a manual builder you can achive a simpler solution by using @Builder annotation, try to search and implement with it.